components/sup/src/event.rs

```diff
@@ -217,14 +218,22 @@ pub fn service_update_started(service: &Service, update: &PackageIdent) {
 /// records how long it took that hook to execute completely.
 pub fn health_check(metadata: ServiceMetadata,
                     check_result: HealthCheckResult,
+                    hook_output: Option<CompleteHookOutput>,
                     execution: Option<Duration>) {
     if stream_initialized() {
         let check_result: types::HealthCheckResult = check_result.into();
+        let exit_status = hook_output.as_ref()
+                                     .and_then(|o| o.get_exit_status().code());
+        let (stdout, stderr) = hook_output.map(CompleteHookOutput::output_streams)
```

Comment:
The naming of `output_streams` here is a little misleading, since we're not really getting streams of anything at this point... we're just capturing whatever output went by.

Reply:
Stream is an overloaded term. I was meaning in the linux [standard stream](https://en.wikipedia.org/wiki/Standard_streams) sense. Not sure of a better term. Maybe `process_stream`?